Steps4Impact/Integration/HealthKitDataProvider.swift

```diff
@@ -34,49 +34,54 @@ class HealthKitDataProvider: PedometerDataProvider {
 
   func retrieveStepCount(forInterval interval: DateInterval,
                          _ completion: @escaping (Result<Int, PedometerDataProvider.Error>) -> Void) {
-    return retrieve(.stepCount, forInterval: interval, completion)
+    return retrieve(.stepCount, forInterval: interval, unit: .count(), completion)
   }
 
   func retrieveDistance(forInterval interval: DateInterval,
                         _ completion: @escaping (Result<Int, PedometerDataProviderError>) -> Void) {
-    return retrieve(.distanceWalkingRunning, forInterval: interval, completion)
+    return retrieve(.distanceWalkingRunning, forInterval: interval, unit: .mile(), completion)
   }
 
-  private func retrieve(_ typeid: HKQuantityTypeIdentifier, forInterval: DateInterval,
+  private func retrieve(_ typeid: HKQuantityTypeIdentifier,
+                        forInterval: DateInterval,
+                        unit: HKUnit,
                         _ completion: @escaping (Result<Int, PedometerDataProvider.Error>) -> Void) {
     guard
       let quantityType = HKSampleType.quantityType(forIdentifier: typeid)
     else { return completion(.failure(.quantityType)) }
-
-    switch HK.authorizationStatus(for: quantityType) {
-    case .sharingDenied:
-      return completion(.failure(.sharingDenied))
-    case .notDetermined:
-      HK.requestAuthorization(toShare: nil, read: [quantityType]) { (success: Bool, error: Error?) in
-        guard success, error == nil else {
-          print("Error getting HealthKit access: \(String(describing: error))")
-          return completion(.failure(.sharingNotAuthorized))
+    
+    HK.getRequestStatusForAuthorization(toShare: [], read: [quantityType]) { [weak self] (status, error) in
```

Comment:
this api is iOS 12+ only and doesn't have the bug that `authorizationStatus(for: quantityType)` has for returning `.sharingDenied` if we don't ask for write permission